client/lib/lib.rs

```diff
@@ -0,0 +1,612 @@
+        session.try_into()?,
+    );
+    RpcCall::new(maybe_rpc_id, node_address, verbose)?.put_deploy(deploy)
+}
+
+/// Make a deploy and output to file or stdout.
+///
+/// * `maybe_rpc_id` is used for RPC-ID as required by the JSON-RPC specification, and is returned
+///   by the node in the corresponding response.  It must be either able to be parsed as a `u32` or
+///   empty.  If empty, a random ID will be assigned.
+/// * `node_address` identifies the network address of the target node's HTTP server, e.g.
+///   `"http://127.0.0.1:7777"`.
+/// * When `verbose` is `true`, the request will be printed to `stdout`.
+/// * `deploy` - Deploy arguments container struct holding options for this deploy. See
+///   `DeployStrParams`.
+/// * `session` - Session arguments container struct holding options for this deploy. See
+///   `SessionStrParams`.
+/// * `payment` - Payment arguments container struct holding options for this deploy. See
+///   `PaymentStrParams`.
+#[allow(clippy::too_many_arguments)]
+pub fn make_deploy(
+    maybe_output_path: &str,
+    deploy: DeployStrParams<'_>,
+    session: SessionStrParams<'_>,
+    payment: PaymentStrParams<'_>,
+) -> Result<()> {
+    Deploy::make_deploy(
+        none_if_empty(maybe_output_path),
+        deploy.try_into()?,
+        payment.try_into()?,
+        session.try_into()?,
+    )
+}
+
+/// Transfers an amount between purses.
+///
+/// * `maybe_rpc_id` is used for RPC-ID as required by the JSON-RPC specification, and is returned
+///   by the node in the corresponding response.  It must be either able to be parsed as a `u32` or
+///   empty.  If empty, a random ID will be assigned.
+/// * `node_address` identifies the network address of the target node's HTTP server, e.g.
+///   `"http://127.0.0.1:7777"`.
+/// * When `verbose` is `true`, the request will be printed to `stdout`.
+/// * `amount` - Specifies the amount to be transferred.
+/// * `maybe_source_purse` - Source purse in the sender's account that this amount will be drawn
+///   from, if it is `""`, it will default to the account's main purse.
+/// * `maybe_target_purse` - Target purse in the sender's account that this amount will be
+///   transferred to.
+/// it is incompatible with `maybe_target_account`.
+/// * `maybe_target_account` - Target account that this amount will be transferred to.
+/// it is incompatible with `maybe_target_purse`.
+/// * `deploy` - Deploy arguments container struct holding options for this deploy.
+#[allow(clippy::too_many_arguments)]
+pub fn transfer(
+    maybe_rpc_id: &str,
+    node_address: &str,
+    verbose: bool,
+    amount: &str,
+    maybe_source_purse: &str,
+    maybe_target_purse: &str,
+    maybe_target_account: &str,
+    deploy: DeployStrParams<'_>,
+) -> Result<JsonRpc> {
+    let target = parsing::get_transfer_target(maybe_target_account, maybe_target_purse)?;
+
+    // transfer always invoke the standard payment
+    let payment = parsing::parse_standard_payment_info(amount)?;
```

Comment:
This is incorrect.  Currently the `transfer` subcommand allows for the full range of payment args, not just standard payment.

Reply:
Ok, thanks for clarifying that - I think I was confused by the presence of two `*amount` arguments - one being `--amount` and the other `--payment-amount`. I'll add the full set of payment args back here.